pkg/netpol/eval/resources.go

```diff
@@ -668,8 +668,35 @@ func (pe *PolicyEngine) createPodOwnersMap() (map[string]Peer, error) {
 }
 
 // GetPeersList returns a slice of peers from all PolicyEngine resources
-// get peers in level of workloads (pod owners) of type WorkloadPeer
+// get peers in level of workloads (pod owners) of type WorkloadPeer, and ip-blocks
+// Deprecated - instead using funcs GetWorkloadPeersList and GetIPBlockPeersLists
 func (pe *PolicyEngine) GetPeersList() ([]Peer, error) {
+	podOwnersMap, err := pe.createPodOwnersMap()
+	if err != nil {
+		return nil, err
+	}
+	_, _, ipBlocks, err := pe.getDisjointIPBlocks()
+	if err != nil {
+		return nil, err
+	}
+
+	// add ip-blocks to peers list
+	res := make([]Peer, len(ipBlocks)+len(podOwnersMap))
+	for i := range ipBlocks {
+		res[i] = &k8s.IPBlockPeer{IPBlock: ipBlocks[i]}
+	}
+	index := len(ipBlocks)
+	// add workload peer objects to peers list
+	for _, workloadPeer := range podOwnersMap {
+		res[index] = workloadPeer
+		index++
+	}
+	return res, nil
```

Comment:
this func can just call GetWorkloadPeersList and GetIPBlockPeersLists, instead of duplicating the code?

Reply:
done